packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -624,20 +1067,24 @@ class HttpRouteSpec extends RouteSpec {
   }
 
   public bind(scope: Construct): RouteSpecConfig {
-    const prefixPath = this.match ? this.match.prefixPath : '/';
-    if (prefixPath[0] != '/') {
-      throw new Error(`Prefix Path must start with \'/\', got: ${prefixPath}`);
+    const prefix = !isMatchPropertiesUndefined(this.match) ? this.match?.pathOrPrefix?.bind(scope).requestMatch.prefix :'/';
```

Comment:
Again, I don't understand what is happening here (we check all properties of `this.match`, but then we only use `this.match.pathOrPrefix`).

Reply:
Please refer [this comment](https://github.com/aws/aws-cdk/pull/15395#discussion_r663271252).